Subject: config.sub: accept substrate as an OS name

Standard one-line config.sub registration so triples like
i386-unknown-substrate are recognised at configure time.

---
diff -urN a/build-aux/config.sub b/build-aux/config.sub
--- a/build-aux/config.sub
+++ b/build-aux/config.sub
@@ -1734,7 +1734,7 @@
 	     | os9* | macos* | osx* | ios* \
 	     | mpw* | magic* | mmixware* | mon960* | lnews* \
 	     | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
-	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
+	     | aos* | aros* | cloudabi* | sortix* | substrate* | twizzler* \
 	     | nindy* | vxsim* | vxworks* | ebmon* | hms* | mvs* \
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
 	     | mirbsd* | netbsd* | dicos* | openedition* | ose* \
